File: packages/twilio_flutter_video/android/src/main/kotlin/com/twilioflutter/video/TwilioFlutterVideoPlugin.kt
package com.twilioflutter.video

import com.twilioflutter.video.bridge.VideoBridge
import com.twilioflutter.video.pigeon.TwilioVideoHostApi
import io.flutter.embedding.engine.plugins.FlutterPlugin

class TwilioFlutterVideoPlugin : FlutterPlugin {
  private var videoBridge: VideoBridge? = null

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    videoBridge = VideoBridge(binding.applicationContext, binding.binaryMessenger)
    TwilioVideoHostApi.setUp(binding.binaryMessenger, videoBridge)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    TwilioVideoHostApi.setUp(binding.binaryMessenger, null)
    videoBridge = null
  }
}

File: packages/twilio_flutter_video/ios/Classes/TwilioFlutterVideoPlugin.swift
import Flutter
import UIKit

public class TwilioFlutterVideoPlugin: NSObject, FlutterPlugin {
  private var bridge: VideoBridge?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let bridge = VideoBridge(binaryMessenger: registrar.messenger())
    TwilioVideoHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: bridge)

    let instance = TwilioFlutterVideoPlugin()
    instance.bridge = bridge
  }
}
